code/code/misc/offense.cc

```diff
@@ -766,332 +766,283 @@ static bool canFleeThisWay(TBeing *ch, dirTypeT dir)
+        return true;
+      }
+    rc = riderAsTBeing->fallOffMount(this, POSITION_SITTING);
     if (IS_SET_DELETE(rc, DELETE_THIS)) {
-      TBeing *tbt = dynamic_cast<TBeing *>(t);
-      if (tbt)
-        tbt->reformGroup();
-      delete t;
-      t = NULL;
+      riderAsTBeing->reformGroup();
+      delete riderAsTBeing;
     }
+    // Don't want reference to rider anymore if they fell off
+    riderAsTBeing = nullptr;
+    panic = true;
   }
-   
-  
-  dirTypeT chosenDir = getDirFromChar(arg);
 
-  soundNumT snd = pickRandSound(SOUND_FLEE_01, SOUND_FLEE_03);
-  playsound(snd, SOUND_TYPE_COMBAT);
+  playsound(pickRandSound(SOUND_FLEE_01, SOUND_FLEE_03), SOUND_TYPE_COMBAT);
 
-  if (!(vict = fight())) {
-    for (i = 0; i < 20; i++) {
-      dirTypeT attempt = dirTypeT(::number(MIN_DIR, MAX_DIR-1));        // Select a random direction 
-      
-      // not fighting, so encourage flight to be in dir PC selected
-      if (chosenDir != DIR_NONE && !panic) {
-	attempt = chosenDir;
-      }
-      if (canFleeThisWay(this, attempt)) {
-        act("$n panics, and attempts to flee.", TRUE, this, 0, 0, TO_ROOM);
-        TBeing *tbt = dynamic_cast<TBeing *>(rider);
-        if (tbt) {
-          act("You turn tail and attempt to run away.", 
-	      TRUE, this, 0, 0, TO_CHAR);
-          loseSneak();
-          iDie = tbt->moveOne(attempt);
-          if (IS_SET_DELETE(iDie, DELETE_THIS)) {
-            tbt->reformGroup();
-            delete tbt;
-            tbt = NULL;
-            return FALSE;
-          }
-          if (iDie == TRUE) {
-            sendTo("You fled head over heels.\n\r");
-            return TRUE;
-          } else {
-            fleeFail(this);
-            return TRUE;
-          }
-        } else {
-          act("You turn tail and attempt to run away.", 
-	      TRUE, this, 0, 0, TO_CHAR);
-          loseSneak();
-          iDie = moveOne(attempt);
-          if (IS_SET_DELETE(iDie, DELETE_THIS))
-            return DELETE_THIS;
-	  
-          if (iDie == TRUE) {
-            sendTo(format("You nearly hurt yourself as you fled madly %swards.\n\r") % dirs[attempt]);
-            REMOVE_BIT(specials.affectedBy, AFF_ENGAGER);
-	    
-            return TRUE;
-          } else {
-            fleeFail(this);
-            return TRUE;
-          }
-        }
-      }
-    }
-    // No exits was found 
+  // Create vector containing all currently valid movement directions
+  std::vector<dirTypeT> validDirections;
+  for (auto direction = MIN_DIR; direction < MAX_DIR; direction++) {
+    if (canFleeThisWay(this, direction)) validDirections.push_back(direction);
+  }
+
+  // If there are no valid flee directions, simply return with the panic
+  // message.
+  if (validDirections.size() == 0) {
     sendTo("PANIC! You couldn't escape!\n\r");
-    return TRUE;
+
+    // Handle troglodyte racial perk
+    if (!::number(0, 1) && getMyRace()->hasTalent(TALENT_MUSK) && getCond(FULL) > 5) {
```

Comment:
Mark this resolved - just a note that I had NO IDEA musk was a thing.

Reply:
Not sure it actually does anything useful, even, but it was used this way previously so I figured we should keep it around. 👍